### ISSUE TITLE
The output of extractor.extract() doesn't correspond to extractor.output_shape()?

### ISSUE THREAD

**Issue by ryangawei** [tqbl/dcase2018_task2#1]
I run the following code:
```
y, sr = librosa.load('./data/train_curated/0f96f548.wav', sr=None)
extractor = LogmelExtractor(sample_rate=sr, n_window=1024, hop_length=512, n_mels=60)
print(extractor.output_shape(y.shape[0]/sr))
melgram = extractor.extract(y)
print(melgram.shape)
```
The output is:
```
(582.0, 60)
(584, 60)
```
I've tried on many .wav files, the result of out output_shape() in 1st dimension is always 2 less than that of extract(). Can you explain why the result is different?

**Comment by tqbl** [comment]
That's a known bug that I admittedly forgot to fix. It doesn't affect the program since the chunk size is fixed.

Anyway, I believe the correct definition of `output_shape()` is
```
n_samples = clip_duration * self.sample_rate
n_frames = n_samples // self.hop_length + 1
return (n_frames, self.mel_fb.shape[0])
```
Can you try and see if you get the correct shape with this?

**Comment by tqbl** [comment]
That's a known bug that I admittedly forgot to fix. It doesn't affect the program since the chunk size is fixed.

Anyway, I believe the correct definition of `output_shape()` is
```
n_samples = clip_duration * self.sample_rate
n_frames = n_samples // self.hop_length + 1
return (n_frames, self.mel_fb.shape[0])
```
Can you try and see if you get the correct shape with this?

**Comment by ryangawei** [comment]
> That's a known bug that I admittedly forgot to fix. It doesn't affect the program since the chunk size is fixed.
> 
> Anyway, I believe the correct definition of `output_shape()` is
> 
> ```
> n_samples = clip_duration * self.sample_rate
> n_frames = n_samples // self.hop_length + 1
> return (n_frames, self.mel_fb.shape[0])
> ```
> 
> Can you try and see if you get the correct shape with this?

It's correct now, thanks for your fast reply!

**Comment by tqbl** [comment]
Fixed in commit 381c8dfc.

**Comment by tqbl** [comment]
Fixed in commit 381c8df.
